src/bootstrap/dylib_util.rs

```diff
@@ -0,0 +1,28 @@
+// Various utilities for working with dylib paths.
+//
+// This file is meant to be included directly to avoid a dependency on the bootstrap library from
+// the rustc and rustdoc wrappers. This improves compilation time by reducing the linking time.
+
+/// Returns the environment variable which the dynamic library lookup path
+/// resides in for this platform.
+pub fn dylib_path_var() -> &'static str {
+    if cfg!(target_os = "windows") {
+        "PATH"
+    } else if cfg!(target_os = "macos") {
+        "DYLD_LIBRARY_PATH"
+    } else if cfg!(target_os = "haiku") {
+        "LIBRARY_PATH"
+    } else {
+        "LD_LIBRARY_PATH"
+    }
+}
```

Comment:
FWIW I think we should at least consider a PR to std adding this as a const in https://doc.rust-lang.org/nightly/std/env/consts/index.html, it feels pretty similar to some of those already exposed there and relatively low cost (and all current tier 1 targets have some notion of the variable, I think).

Reply:
Windows doesn't have a separate `LD_LIBRARY_PATH` env var, but reuses `PATH` for both spawning executables and loading dlls.